sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy

```diff
@@ -786,27 +787,8 @@ class ContainerAPITest extends APISpec {
         response.getStatusCode() == 404
     }
 
-    def "Delete if exists service container min"() {
-        setup:
-        def containerName = generateContainerName()
-        primaryBlobServiceClient.createBlobContainer(containerName)
-
-        when:
-        def result = premiumBlobServiceClient.deleteBlobContainerIfExists(containerName)
-
-        then:
-        !premiumBlobServiceClient.getBlobContainerClient(containerName).exists()
-        result
-    }
-
-    def "Delete if exists service container that does not exist"() {
-        when:
-        def response = premiumBlobServiceClient.deleteBlobContainerIfExistsWithResponse(generateContainerName(), null)
-
-        then:
-        response.getStatusCode() == 404
-    }
-
+    // We can't guarantee that the requests will always happen before the container is garbage collected
```

Comment:
the other option is to refactor this test (and other similar tests) to poll until container gets GC'ed for some period of time before executing the api under test.

Reply:
Btw. do we return `true` when we get 409 with `ContainerBeingDeleted` ? I would think we should return `false` in that case.

Reply:
It's the opposite problem for this test. The test is checking the behavior of issuing the request while the container is still being GC'ed. It looks like the live pipelines are sometimes waiting too long and not getting the expected response as a result. I don't think we can guarantee the second request comes within a particular time frame, so the most consistent thing to me seems to be to make it playback only. 

Let me run the pipeline before that commit, though, and double check what the behavior is.

Reply:
do other APIs still work while container is GC'ed? if not then I'd return `false` from the API and consider the container as "deemed to be gone".
Also looking from another angle - if customer issues multiple calls to this API then only first should return `true` because it's the one that actually triggers deletion.

Reply:
Looks like we may actually have different behavior between sync and async.
[Async](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerAsyncClient.java#L605) is statusCode != 404
[Sync](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobContainerClient.java#L453) is statusCode == 202

That said, the [Rest docs](https://docs.microsoft.com/en-us/rest/api/storageservices/delete-container#remarks) seem to state that 409 only comes with creating a container not yet GC'ed and that deleting twice should always 404, so in effect these may be the same.

Reply:
sounds like a bug then. I think Sync is doing right thing.

Reply:
I'll update async

Reply:
@kasobol-msft Doing some experimenting, it looks like the service actually returns 202 during the GC phase for the record. We can still keep the logic to check that the status is 202, but FYI.

I also think it's best to still keep those tests as playback only because we can't guarantee that the second request comes within the GC period, and I believe that's what's tripping up the live tests on this PR

Reply:
sounds good.